Elevator/ElevatorControlPanel.lua

```diff
@@ -14,37 +14,144 @@ function drawButton(scr)
   scr.write(scr.text)
 end
 
+function drawFloors()
+	mon.clear()
+	--for each line on screen
+	for i = 0, 10 do -- 10 is line limit of one monitor
+		--only process defined buttons
+		if(floors[i]) then
+			--render
+			drawButton(floors[i])
+		end
+	end
+end
+
+function MonitorEvent(event) -- Select floors etc.
+	if floors[event[4]] then-- Click Y
+		for i = 0, 10 do -- 10 is line limit of one monitor
+			--only process defined buttons
+			if(floors[i]) then
+				floors[i].text = floors[i].text:gsub(">>","")
+			end
+		end
+	
+		--add chevrons to whichever button was hit
+		floors[event[4]].text = ">>"..floors[event[4]].text
+		lastButtonHit = event[4]
+		drawFloors()
+	else
+		return
+    end
+end
+
+function ColorPicker()
+	mon.clear()
+	local _colors = {colors.orange, colors.lightBlue, colors.yellow, colors.pink, colors.cyan, colors.purple, colors.blue, colors.green, colors.red, colors.black}
```

Comment:
Would be good to make this array global as we also use it in FloorPicker.
It might not be necessary as colors have number codes already http://www.computercraft.info/wiki/Colors_(API)#Colors
, could maybe just use those instead.

Reply:
Can make the color picker an outside module entirely, defining color lists and the colorpicker() withion